tests/perf-test/SignInPage.perf-test.tsx

```diff
@@ -32,10 +32,6 @@ jest.mock('@react-navigation/native', () => {
             navigate: jest.fn(),
             addListener: () => jest.fn(),
         }),
-        createNavigationContainerRef: () => ({
-            addListener: () => jest.fn(),
-            removeListener: () => jest.fn(),
-        }),
```

Comment:
More context on Slack: https://expensify.slack.com/archives/C01GTK53T8Q/p1706732738472299

SignInPage reassure test was failing with this error: 

```
  ● SignInPage › [SignInPage] should add username and click continue button

    TypeError: _Navigation.navigationRef.isReady is not a function

      37 |
      38 | const triggerUnreadUpdate = () => {
    > 39 |     const currentReportID = navigationRef.isReady() ? Navigation.getTopmostReportId() ?? '' : '';
         |                                           ^
      40 |
      41 |     // We want to keep notification count consistent with what can be accessed from the LHN list
      42 |     const unreadReports = getUnreadReportsForUnreadIndicator(allReports, currentReportID);
```

And I was told to remove this part as:

> I recently changed it to solve another issue, but it looks like we don't need it at all.

Reply:
On recent sync w/ main the `createNavigationContainerRef` was added back, but tests on this PR were still failing since the `navigationRef.getState()` mock was missing, I added the mock and they still failed because of this line:

https://github.com/Expensify/App/blob/3a702ce30137d84275dc56a7dd472718672652a9/src/libs/Navigation/getTopmostReportId.ts#L19

where state.routes was undefined and .filter was failing. I added optional chaining there and they pass now.